### PR TITLE
improve efficiency of metrics lookups and font creation by caching

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

File: src/mpc-hc/CMPCThemeUtil.h [hunk 1]
```diff
@@ -54,7 +55,7 @@ class CMPCThemeUtil
     static CSize GetTextSizeDiff(CString str, HDC hDC, CWnd* wnd, int type, CFont* curFont);
 
 
-    static void GetMetrics(NONCLIENTMETRICS* ncMetrics);
+    static void GetMetrics(bool reset = true);
```
Comment (clsid2) [inline src/mpc-hc/CMPCThemeUtil.h]: default value of reset should be `false`?

File: src/mpc-hc/CMPCThemeMenu.cpp [hunk 2]
```diff
@@ -503,7 +521,8 @@ void CMPCThemeMenu::GetStrings(MenuObject* mo, CString& left, CString& right)
 
 void CMPCThemeMenu::MeasureItem(LPMEASUREITEMSTRUCT lpMeasureItemStruct)
 {
-    initDimensions();
+    initDimensions(); //should happen before drawitem
```
Comment (clsid2) [inline src/mpc-hc/CMPCThemeMenu.cpp:522]: Are these initDimensions calls redundant? Maybe use `assert(font.m_hObject)` instead?

File: src/mpc-hc/CMPCThemeUtil.h (hunk 1 above)
Comment (adipose) [reply src/mpc-hc/CMPCThemeUtil.h]: fixed.

File: src/mpc-hc/CMPCThemeMenu.cpp (hunk 2 above)
Comment (adipose) [reply src/mpc-hc/CMPCThemeMenu.cpp:522]: initDimensions gets called on first MeasureItem, and shouldn't need to be called again.  But out of paranoia I initialized it in any function that can use the font object (including drawitem).  I think it's impossible for drawitem to be called first, but I am not 100% sure.

I added an if block to be sure, now.

**Pull request merged**